packages/hub/src/lib/parse-safetensors-metadata.ts

```diff
@@ -14,7 +14,7 @@ export const SAFETENSORS_INDEX_FILE = "model.safetensors.index.json";
 /// but in some situations safetensors weights have different filenames.
 export const RE_SAFETENSORS_FILE = /\.safetensors$/;
 export const RE_SAFETENSORS_INDEX_FILE = /\.safetensors\.index\.json$/;
-export const RE_SAFETENSORS_SHARD_FILE = /\d{5}-of-\d{5}\.safetensors$/;
+export const RE_SAFETENSORS_SHARD_FILE = /(?<shard>\d{5})-of-(?<total>\d{5})\.safetensors$/;
```

Comment:
Might as well add the "prefix"? If you're planning on extracting it

Reply:
in that case add `[_-]` between the captured prefix and the shard number

(most models on the hub use a dash at this position (it's what's generated by transformers), but not `bigscience/bloom`)